crates/server/src/admin/utils/auth.rs

```diff
@@ -131,26 +131,36 @@ pub fn verify_node_signature(
             let chain_id = network_metadata.chain_id.clone();
 
             let result = match wallet_name.as_str() {
-                "argentX" => tokio::task::block_in_place(|| {
-                    tokio::runtime::Runtime::new()
-                        .unwrap()
-                        .block_on(verify_argent_signature(
-                            message_hash,
-                            signature,
-                            wallet_metadata.signing_key.clone(),
-                            &payload.message.message,
-                            &rpc_node_url,
-                            &chain_id,
-                        ))
-                }),
-                "metamask" => verify_metamask_signature(
-                    message_hash,
-                    signature,
-                    wallet_metadata.signing_key.clone(),
-                    &payload.message.message,
-                    wallet_metadata.wallet_address.clone().unwrap_or_default(),
-                    &chain_id,
-                ),
+                "argentX" => {
+                    verify_argent_signature(
+                        message_hash,
+                        signature,
+                        &wallet_metadata.signing_key.clone(),
+                        &payload.message.message,
+                        &rpc_node_url,
+                        &chain_id,
+                    )
+                    .await
+                }
+                "metamask" => {
+                    let wallet_address: &str = match &wallet_metadata.wallet_address {
+                        Some(s) => s,
+                        None => {
+                            return Err(ApiError {
+                                status_code: StatusCode::BAD_REQUEST,
+                                message: "Wallet address not present.".into(),
+                            })
+                        }
+                    };
```

Comment:
style: with the let-else syntax, you can rewrite as:

```suggestion
                    let Some(wallet_address) = &wallet_metadata.wallet_address else {
                        return Err(ApiError {
                            status_code: StatusCode::BAD_REQUEST,
                            message: "Wallet address not present.".into(),
                        })
                    };
```

up to you tho